display_list/display_list_builder.cc

```diff
@@ -540,6 +555,7 @@ void DisplayListBuilder::transform2DAffine(
       SkScalarsAreFinite(mxt, myt) &&
       !(mxx == 1 && mxy == 0 && mxt == 0 &&
         myx == 0 && myy == 1 && myt == 0)) {
+    checkForDeferredSave();
```

Comment:
You missed the transformFullPerspective method - also needs a checkForDeferred (but only in the case where it pushes its own record, the other case will transfer control to transform2DAffine which already does the check as needed).

Reply:
Also transformReset() should trigger the checkForDeferred as well.